pkg/api/v1alpha1/thirdparty/tinkerbell/workflow_types.go

```diff
@@ -0,0 +1,34 @@
+// Package represents https://pkg.go.dev/github.com/tinkerbell/tink@v0.6.0/workflow#pkg-types with json tags.
+//+kubebuilder:object:generate=true
+package tinkerbell
+
```

Comment:
Why recreate these types vs just using existing types from https://github.com/tinkerbell/tink/blob/main/workflow/types.go?

Reply:
So two reasons actually.
1. k8s CRD definition require json tags on structs, or else `controller-gen` tool fails.
2. We also needed  DeepCopy method definitions for those types.

hence I had to redefine them. I thought about maybe upstreaming the changes by adding the tags :)

Reply:
Ok cool thanks for background.